subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -1122,22 +1145,41 @@ static int scan_delegator_rem_src(struct bt_conn *conn,
 	state = &internal_state->state;
 
 	/* If conn == NULL then it's a local operation and we do not need to ask the application */
-	if (conn != NULL && (state->pa_sync_state == BT_BAP_PA_STATE_INFO_REQ ||
-			     state->pa_sync_state == BT_BAP_PA_STATE_SYNCED)) {
-		/* Unlock mutex to avoid potential deadlock on app callback */
-		err = k_mutex_unlock(&internal_state->mutex);
-		__ASSERT(err == 0, "Failed to unlock mutex: %d", err);
+	if (conn != NULL) {
 
-		/* Terminate PA sync */
-		err = pa_sync_term_request(conn, &internal_state->state);
-		if (err != 0) {
-			LOG_DBG("PA sync term from %p was rejected with reason %d", (void *)conn,
-				err);
+		if (scan_delegator_cbs != NULL &&
+			scan_delegator_cbs->remove_source != NULL) {
```

Comment:
```suggestion
		if (scan_delegator_cbs != NULL && scan_delegator_cbs->remove_source != NULL) {
```

Maximum column width is 100 in Zephyr